### ISSUE TITLE
Urls targeting collections of EdmComplexObjects fail to serialize.

### ISSUE THREAD

**Issue by mtroth-microsoft** [OData/WebApi#1597]
if I target the entity set:
/Entities/key
and the entity type contains a Collection(ComplexType) property, the result works as expected. But when I navigate directly to that property, the service errors out because the media formatter is unable to find a qualifying serializer to handle List<EdmComplexObject>.
/Entities/key/ListOfComplexObjects

this was tested against master.

**Comment by xuzhg** [comment]
@mtroth-microsoft I don't know why did you get such error. It should work for the current ASP.NET OData library.

![image](https://user-images.githubusercontent.com/9426627/44885597-8dcde700-ac76-11e8-8f19-ee743cb3b544.png)

Did you implement the correct method in your entityset controller? In the convention routing, the method name should be : `Get{ComplexPropertyName}`

**Comment by mtroth-microsoft** [comment]
Yes. Are you sure yours is a typeless model? I have no Clr classes anywhere in my component.

**Comment by xuzhg** [comment]
Yes. It should work for untyped (typeless) model.
For example, I have the following untyped model. (Without CLR type).
```C#
        public static IEdmModel GetUntypedModel()
        {
            var model = new EdmModel();
            var customer = new EdmEntityType("NS", "Customer");
            customer.AddKeys(customer.AddStructuralProperty("Id", EdmPrimitiveTypeKind.Int32));
            model.AddElement(customer);

            var address = new EdmComplexType("NS", "Address");
            address.AddStructuralProperty("City", EdmPrimitiveTypeKind.String);
            model.AddElement(address);
            customer.AddStructuralProperty("Addresses", new EdmCollectionTypeReference(new EdmCollectionType(new EdmComplexTypeReference(address, true))));
            var container = new EdmEntityContainer("NS", "Default");
            var customers = new EdmEntitySet(container, "Customers", customer);
            model.AddElement(container);
            container.AddElement(customers);

            return model;
        }
```

In the controller, I have:


```C#

IEdmCollectionTypeReference edmType;
var model = Request.GetModel();
var address = model.SchemaElements.OfType<IEdmComplexType>().First();
edmType = new EdmCollectionTypeReference(new EdmCollectionType(new EdmComplexTypeReference(address, true)));
EdmComplexObjectCollection collection = new EdmComplexObjectCollection(edmType);

EdmComplexObject complex = new EdmComplexObject(address);
complex.TrySetPropertyValue("City", "RedmondUntyped");
collection.Add(complex);

complex = new EdmComplexObject(address);
complex.TrySetPropertyValue("City", "BellevueUntyped");
collection.Add(complex);

return Ok(collection);
```

I can get:

![image](https://user-images.githubusercontent.com/9426627/44935466-ae08af00-ad25-11e8-909f-83bc92344da5.png)

**Comment by mtroth-microsoft** [comment]
what serializer are you using?

**Comment by xuzhg** [comment]
@mtroth-microsoft I did not change or and anything. I think I used the default serializer.

**Comment by mtroth-microsoft** [comment]
probably a difference in how the models are created then. I have created a PR against this item so you can see what is needed to make the scenario work for some who is loading their model directly off of a CSDL and returning both entity sets and sets of complex objects that have no type.

**Comment by mtroth-microsoft** [comment]
I don’t need this. I have a repro of the problem, Sam.
I have submitted a PR that addresses the issue.

**Comment by xuzhg** [comment]
@mtroth-microsoft The PR #1607 looks you want to process the serialization of IEnumerable<IEdmEntityObject> or IEnumerable<IEdmComplexObject>, right?

**Comment by mikepizzo** [comment]
Yes; I think the difference is in @xuzhg's sample, even though CLR types are not used, the Edm type "Address" is defined.  In @mtroth-microsoft's case there is no Edm type defined for Address -- the collection of complex type uses Collection(Edm.ComplexType).

**Comment by xuzhg** [comment]
@mikepizzo Now, I see.  Just like the property named "Addresses".

```xml
<EntityType Name="Customer" >
     ...
    <Property Name="Addresses" Type="Collection(Edm.ComplexType)" />
</EntityType>
```

**Comment by KanishManuja-MS** [comment]
@xuzhg @mikepizzo As discussed, the correct fix for this issue should be to update EdmEntityCollection to handle collection of Edm.complextype and Edm.enititytype, Edm.Primitive, Edm.Untyped.
